### ISSUE TITLE
`runOnNewVersion` saves version numbers with non-numerical values and does not sanitize them upon retrieval

### ISSUE THREAD

**Issue by Camerash** [MostafaSolimanMO/once#24]
In version `1.5.4` the logic for version number retrieval in `runOnNewVersion` is:
```dart
final savedVersion = preferences.getString(onceKey) ?? '';
String existingVersion = currentVersion.replaceAll(RegExp(r'[^\d]+'), '');
```

Where as in `1.5.3` it looks like:
```dart
final savedVersion = preferences.getString(onceKey)!.replaceAll(".", "");
String existingVersion = currentVersion.replaceAll(".", "");
```

In the newer version `1.5.4` the sanitation for removing non-numerical in `savedVersion` is removed.

However within `runOnNewVersion` the library is saving also non-sanitized version number to `SharedPreference`:
```dart
String currentVersion = packageInfo.version;
/// ...
/// ...
/// ...
preferences.setString(onceKey, currentVersion);
```

Therefore on version check it will still throw error:
```
flutter: FormatException: 2.3.4
flutter: 
#0      num.parse (dart:core/num.dart:574:5)
#1      OnceRunner.runOnNewVersion (package:once/src/runner/runner.dart:166:44)
<asynchronous suspension>
```

I suggest sanitizing `savedVersion` variable with the same logic as that of `existingVersion`, this will also ensure value saved using previous version of `once` will decode version information correctly.

## System Info:
`flutter`: `3.13.6`
`once`: `1.5.4`
`package_info_plus`: `4.1.0`

**Comment by Camerash** [comment]
Please let me know if this issue is valid, and if yes I can help contribute and put a PR up. Thanks!

**Comment by MostafaSolimanMO** [comment]
Thank you so much for this, Esmond.
Yes, it's valid. I've fixed it and I'm publishing a new release

**Comment by MostafaSolimanMO** [comment]
Fixed on [1.6.0](https://github.com/MostafaSolimanMO/once/releases/tag/1.6.0).
Thanks @Camerash
